honeybee/radiance/parameters/_parametersbase.py

```diff
@@ -210,13 +210,23 @@ def _parse_rad_parameters(self, parameters_string):
         if parameters_string is None:
             return rad_par
 
-        # split input string: each part will look like key value (eg ad 1)
-        par_list = parameters_string.split("-")
+        # use re to find the start and end index for each parameter in parameter string
+        indices = [(m.start(0), m.end(0)) for m in re.finditer(r'-[a-zA-Z]{1,6}'\
```

Comment:
Do we really need `{1, 6}` here? If it is one or more we can do: `-[a-zA-Z]+`. This looks like a one or many case but 6 can end up being tricky.

If we're going the regex path we can do something like this to get the key, values as a group:

```Python
import re

pattern = r'-([a-zA-Z]+)\s([-+]*\d+[eE][-+]\d+|\d*)'
parameters_string = '-lw 2e-06 -I -ab 6 -ad 1000 -aa 1E-06 -ad +2E+08'
results = re.finditer(pattern, parameters_string.strip())

# now each group is the key and value that we are looking for 
for res in results:
    print(res.groups())

>> ('lw', '2e-06')
>> ('I', '')
>> ('ab', '6')
>> ('ad', '1000')
>> ('aa', '1E-06')
>> ('ad', '+2E+08')
```
I'm sure people will do crazy things that this regex fails to address but it should work for most of the cases. This is one of those cases that we can enhance as we go.